src/runtime-rs/crates/resource/src/rootfs/block_rootfs.rs

```diff
@@ -67,8 +68,22 @@ impl BlockRootfs {
         let mut device_id: String = "".to_owned();
         if let DeviceType::Block(device) = device_info {
             storage.driver = device.config.driver_option;
-            storage.source = device.config.virt_path;
             device_id = device.device_id;
+
+            // TODO: The following would work for blk-pci and mmio drivers.
+            // The source for storage drivers needs to be updates for other drivers like
+            // VIRTIO_SCSI, VIRTIO_BLK_CCW and VIRTIO_PMEM once support for them is added.
+            if block_driver == VIRTIO_BLK_PCI {
```

Comment:
There will be more drivers to be added here, right ?
if so, IMO, we should take the future possible driver into account.
And I prefer match xx {}, instead if else here.

Reply:
@Apokleos - I understand your view. However, @amshinde did mention that there will need to be future changes at the top of the PR. A bit more detail...

- This change fixes the specific (and fairly urgent - see below) issue of handling devicemapper block (DM) devices, which currently does not work in runtime-rs.
- That bug is stopping us from enabling the Cloud Hypervisor CI (see #9034).
- We need to have the runtime-rs CH CI enabled as soon as possible to:
  - protect the code in that driver.
  - allow us to demonstrate that the runtime-rs CH driver is on parity with the golang runtime's CH driver.
- Hence, I don't think we should stop this PR landing until every possible block device type (some of which are even architecture specific) is also added to it.

How about we raise issues for the remaining device types to make it clear to the community what further work needs to be done in this area?

Reply:
@jodh-intel I agree to push the PR merged ASAP.